src/main/java/com/uber/cadence/internal/metrics/ClientVersionEmitter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.metrics;
+
+import com.uber.cadence.internal.Version;
+import com.uber.m3.tally.Scope;
+import com.uber.m3.util.ImmutableMap;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Properties;
+
+public class ClientVersionEmitter implements Runnable {
+
+  private final Scope metricScope;
+  private final Map<String, String> tags;
+
+  public ClientVersionEmitter(Scope metricScope, String domain) {
+    if (metricScope == null) {
+      this.metricScope = NoopScope.getInstance();
+    } else {
+      this.metricScope = metricScope;
+    }
+
+    if (domain == null) {
+      domain = "";
+    }
+
+    Properties prop = new Properties();
+    InputStream in = Version.class.getResourceAsStream("/com/uber/cadence/version.properties");
```

Comment:
What's the process whereby we update this?

Reply:
It is generated by gradle, so no need to update manually